jupyterhub/app.py

```diff
@@ -2668,6 +2675,24 @@ async def start(self):
             self.log.error("Failed to bind hub to %s", self.hub.bind_url)
             raise
 
+        # start the metrics server
+        if self.metrics_port is not None:
+            metrics_tornado_application = web.Application(
+                metricshandler.default_handlers, **self.tornado_settings
+            )
+            self.metrics_server = tornado.httpserver.HTTPServer(
+                metrics_tornado_application,
+                ssl_options=ssl_context,
+                xheaders=True,
+                trusted_downstream=self.trusted_downstream_ips,
+            )
+            try:
+                self.metrics_server.listen(self.metrics_port, address="::")
```

Comment:
I don't think we want to unconditionally bind on all interfaces. Instead, let's use the hub ip already. If that's not always appropriate, then it probably makes sense to use `metrics_url` instead of `metrics_port` so we can configure both port and interface.

```suggestion
                self.metrics_server.listen(self.metrics_port, address=bind_url.hostname)
```